### PR TITLE
ENG-4664 feat(portal,1ui): update queries in stake modal

### PR DESCRIPTION
**Pull request by Vitalsine85**
## Affected Packages

Apps

- [ ] data populator
- [x] portal
- [ ] template

Packages

- [x] 1ui
- [ ] api
- [ ] graphql
- [ ] protocol
- [ ] sdk

Tools

- [ ] tools

## Overview

Based off branch/PR 4383 since there were some important updates to the stake modal that had to be accounted for. Cannot merge until we get that PR merged in.

- Swapped out vaultDetails fetcher for react query hook
- honestly can't remember some of the other updates since so much is crammed into this due it being based off another branch. Once that is merged and this is rebased, it will clear that all up.

## Screen Captures

If applicable, add screenshots or screen captures of your changes.

## Declaration

- [x] I hereby declare that I have abided by the rules and regulations as outlined in the [CONTRIBUTING.md](https://github.com/0xIntuition/intuition-ts/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jonathanprozzi** [comment]
@Vitalsine85 -- I removed the `do not merge` label since this is onto the feature branch

**Review by jonathanprozzi — COMMENTED** [review]
Lmk what you think about the queryKeys comment!

File: apps/portal/app/lib/hooks/mutations/useStakeMutation.tsx [hunk 1]
```diff
@@ -71,8 +71,17 @@ export function useStakeMutation(contract: string, mode: 'deposit' | 'redeem') {
           value: actionType === 'buy' ? parsedValue : undefined,
         })
       },
-      onSuccess: async () => {
-        queryClient.invalidateQueries({ queryKey: ['get-stats'] })
+      onSuccess: async (_, variables) => {
+        await queryClient.invalidateQueries({
```
[the remark this answers is not in the document]
Comment (Vitalsine85) [reply apps/portal/app/lib/hooks/mutations/useStakeMutation.tsx:75]: I can't remember honestly. I will give it a try this morning and update.

[the remark this answers is not in the document]
Comment (Vitalsine85) [reply apps/portal/app/lib/hooks/mutations/useStakeMutation.tsx:75]: Had to make a couple adjustments but got it working, good callout!

**Review by jonathanprozzi — APPROVED** [review]

**Pull request merged**